poh/src/poh_recorder.rs

```diff
@@ -264,6 +265,15 @@ pub struct WorkingBank {
     pub min_tick_height: u64,
     pub max_tick_height: u64,
     pub transaction_index: Option<usize>,
+    pub entry_index: Option<usize>,
+}
+
+impl WorkingBank {
+    fn increment_entry_index(&mut self) {
+        if let Some(index) = self.entry_index.as_mut() {
+            *index += 1
+        }
+    }
```

Comment:
I'm also thinking maybe it would be better to just track all these indexes in `tpu_entry_notifier`? I.e. if you see a new entry for the same slot, then bump the index. This has a few benefits

1. would allow us to leave `poh_recorder` unchanged and unaware of this additional logic
2. if the track entry option is off, we can avoid spinning up the tpu_entry_notifier service altogether, and no additional work is incurred

Reply:
I like it. Thanks for the idea

Reply:
Actually, there is an additional reason for tracking the entry indexes in poh_recorder: in order to be able to figure out which transactions fell in which entry, we are planning to add the entry index to the [geyser transaction notification](https://github.com/solana-labs/solana/blob/40d3e8e8fad79ceda31c06446f50dd3061cd5a16/geyser-plugin-interface/src/geyser_plugin_interface.rs#L139). The easiest way to do that is to include the entry index with the transaction-starting-index that's returned [here](https://github.com/solana-labs/solana/blob/40d3e8e8fad79ceda31c06446f50dd3061cd5a16/poh/src/poh_recorder.rs#L843)